presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -4119,22 +4137,31 @@ public void testPredicatePushDownToTableScan()
         }
     }
 
-    @Test
-    public void testParquetTimestampPredicatePushdown()
+    @DataProvider
+    public Object[][] timestampPrecisionAndValues()
+    {
+        // TODO: revisit values once we handle write path
+        return new Object[][] {{3, "2012-10-31 01:00:08.123"}, {6, "2012-10-31 01:00:08.123000"}, {9, "2012-10-31 01:00:08.123000000"}};
+    }
+
+    @Test(dataProvider = "timestampPrecisionAndValues")
+    public void testParquetTimestampPredicatePushdown(int precision, String value)
     {
+        Session session = timestampPrecisionSession(precision);
+        assertUpdate("DROP TABLE IF EXISTS test_parquet_timestamp_predicate_pushdown");
         assertUpdate("CREATE TABLE test_parquet_timestamp_predicate_pushdown (t TIMESTAMP) WITH (format = 'PARQUET')");
-        assertUpdate("INSERT INTO test_parquet_timestamp_predicate_pushdown VALUES (TIMESTAMP '2012-10-31 01:00')", 1);
-        assertQuery("SELECT * FROM test_parquet_timestamp_predicate_pushdown", "VALUES (TIMESTAMP '2012-10-31 01:00')");
+        assertUpdate(format("INSERT INTO test_parquet_timestamp_predicate_pushdown VALUES (TIMESTAMP '%s')", value), 1);
+        assertQuery(session, "SELECT * FROM test_parquet_timestamp_predicate_pushdown", format("VALUES (TIMESTAMP '%s')", value));
 
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(
-                getSession(),
-                "SELECT * FROM test_parquet_timestamp_predicate_pushdown WHERE t < TIMESTAMP '2012-10-31 01:00'");
+                session,
+                format("SELECT * FROM test_parquet_timestamp_predicate_pushdown WHERE t < TIMESTAMP '%s'", value));
```

Comment:
We need to make sure the equalities and inequalities are properly applies when the values being read are rounded.

It will be easier to validate when we support writes, but even then this may depend on writer's "skills" when creating file statistics, so we should have a product test as well.

We need these for both ORC and Parquet